carmin-server/server/resources/path.py

```diff
@@ -30,47 +35,66 @@ def get(self, complete_path: str = ''):
         instead, where `schema` is the Schema of the class to be returned.
         """
 
-        action = request.args.get('action', '')
         data_path = app.config['DATA_DIRECTORY']
-        requested_data_path = os.path.realpath(
+        action = request.args.get('action', '')
```

Comment:
Instead of default value, could use a None validation first (get returns None in case the arg is missing) to return a different error in case of a missing action and invalid action.

Reply:
However, this insures some of your future validations with your current logic, so it is correct still.

Reply:
I like this suggestion. Thanks